nengo/tests/test_simulator.py

```diff
@@ -99,3 +103,15 @@ def test_close_steps(RefSimulator):
         sim.run_steps(1)
     with pytest.raises(SimulatorClosed):
         sim.step()
+
+
+def test_warn_on_opensim_gc(Simulator):
+    with nengo.Network() as net:
+        nengo.Ensemble(10, 1)
+
+    sim = Simulator(net)
+    assert sim
+
+    with warns(ResourceWarning):
+        sim = None
+        gc.collect()
```

Comment:
I think even with `del sim` the call to `gc.collect()` is necessary because `del` doesn't call `__del__` directly.